helix-term/src/ui/picker.rs

```diff
@@ -112,9 +112,9 @@ impl Preview<'_, '_> {
     /// Alternate text to show for the preview.
     fn placeholder(&self) -> &str {
         match *self {
-            Self::EditorDocument(_) => "<File preview>",
+            Self::EditorDocument(_) => "<Invalid File Location>",
             Self::Cached(preview) => match preview {
-                CachedPreview::Document(_) => "<File preview>",
+                CachedPreview::Document(_) => "<Invalid File Location>",
```

Comment:
Why the change? Casing should match the rest of the variants

Reply:
I fixed the casing.


These branches were actually unreachable so far. We only show the placeholder if `preview.document()` returns `None` (which doens't happen for these two cases:

``` rust
    fn document(&self) -> Option<&Document> {
        match self {
            Preview::EditorDocument(doc) => Some(doc),
            Preview::Cached(CachedPreview::Document(doc)) => Some(doc),
            _ => None,
        }
    }
```

However, with this PR we also show the placeholder if `range` is invalid so the only way to endup in this branch is if the location provided by whatever the picker is previewing is invalid (usually the LSP but for global search the file could also have been changed by another process since the initial search ran).

A way to make this a bit cleaner would be to `document()` to have the following signature:


``` rust
 /// Returns the document to preview if available or a placeholder string otherwise
 fn document(&self) -> Result<&Document, &str>
```

but I wanted to keep the change minimal here.